packages/core-components/src/components/AlertDisplay/AlertDisplay.tsx

```diff
@@ -57,6 +62,14 @@ export function AlertDisplay(props: AlertDisplayProps) {
     };
   }, [alertApi]);
 
+  useEffect(() => {
+    const [current] = messages;
+    if (current && current.display === 'transient')
+      setTimeout(() => {
```

Comment:
This timeout needs to be cancelled if the component is unmounted. You'll also run into some race conditions with multiple transient messages at once. I'd say you'll need to add a couple of tests to make sure that's all working as intended.

Reply:
Yes, for sure. I'm just not sure how to approach the tests, could you maybe point me to some examples? That would help me get rolling

Reply:
You probably want to use mock timers to make sure that the timeouts happen correctly without actual waiting, which tends to be very flaky. See for example https://github.com/backstage/backstage/blob/6752df627924041300e3b3ea3e3de7779e7cc9f1/plugins/search-react/src/components/SearchBar/SearchBar.test.tsx#L122

It might be easier to refactor so that the messages are provided by their own hook, letting you test the hook directly rather than the entire component, for example https://github.com/backstage/backstage/blob/6752df627924041300e3b3ea3e3de7779e7cc9f1/plugins/catalog-react/src/hooks/useEntityListProvider.test.tsx#L122

Reply:
Slowly working away at this, tests have always been a bit of a challenge for me so thanks for the pointers 👍

Reply:
Yep it's a tricky one for sure. You can also push some WIP changes if you want me to have a look.

Reply:
OK, this ended up being easier than I thought, usually ends up that way for me with tests. Anyway I went the mock timers route as the refactor just wasn't coming to me. This seems to work well. I do have one test that I'm not sure how to setup. Could use your help so will call that out.

Also, I made sure to clear the timeout at the end. 👍